src/lib/viewers/doc/ImagePageLoader.js

```diff
@@ -0,0 +1,89 @@
+import EventEmitter from 'events';
+import RepStatus from '../../RepStatus';
+
+const REPS = ['png_1024x1024', 'png_2048x2048', 'jpg_1024x1024'];
+
+class ImagePageLoader extends EventEmitter {
+    /**
+     * [constructor]
+     *
+     * @param {Object} options Viewer options
+     */
+    constructor(options) {
+        super();
+        this.availableReps = {};
+        this.options = options;
+        this.checkAvailableReps();
+    }
+
+    /**
+     * Gets a representation based on preview size
+     *
+     * @param {number} previewHeight - Height of preview window
+     * @param {number} previewWidth - Width of preview window
+     * @return {Object} Instance of RepStatus
+     */
+    getRepresentation(previewHeight, previewWidth) {
+        let rep;
+        let size;
+        const maxDimension = Math.max(previewHeight, previewWidth);
+        Object.keys(this.availableReps).forEach((repType) => {
```

Comment:
Could pull out the enclosed logic into it's own function? `getLargestAllowedRep()` or something like that? Also, if you want the first one, and not to do this check through the whole array, you could use `Array.prototype.some()` instead of `forEach()`